### DISCUSSION TITLE
Is there a way to turn off power failover if RPi has been manually halted?

### DISCUSSION THREAD

**Discussion by mbstechnologiessales** [PiSupply/PiJuice#782] [Q&A] [answered]
Hi, 
I am running what will be a headless remote RPi-4B used to run continuous video playback, but will have power cut to the building nightly. I have been testing the power failover features for allowing safe shutdown on "no power" system event using SYS_FUNC_HALT_POW_OFF, with great success. But, there will be times I will need to remotely shutdown the system during the day while building power is still active, leaving the red LED lit since the USB-C cord is still plugged in to the RPi supplying 5v. The system will not be restarted until the next day in these instances when building power is restored. This is fine until the building power is shutoff at night and the RPi red LED turns off for a second and then back on as the PiJuice starts supplying battery power (when the RPi was in a manually halted state) and then drains the battery. It restarts in the morning just fine otherwise once the building power is restored. Is there a way for the PiJuice to stop failing over to battery power only if the RPi has been manually halted? Maybe something similar to the "heartbeat" the watchdog uses for resetting the RPi, but instead realizing the RPi is halted, and not supplying battery power unnecessarily, allowing restoration of building power in the morning to wake the RPi?

Thank you,

Michael

**Comment by tvoverbeek** [comment, marked as answer]
Use the 'Software Halt Power Off' option in the system task menu (https://github.com/PiSupply/PiJuice/tree/master/Software#system-task-menu for the GUI or its equivalent in the CLI). This will turn off the the 5V from the PiJuice to the RPi when you do a manual shutdown. Hence no problem when the building power goes off.

**Reply by mbstechnologiessales** [reply]
Thank you for the suggestion and quick response. I thought that would do it as well. I tried that and set the delay to 60 seconds to give the pi time to shutdown. The pi's red light will stay lit after everything is halted, even after mains power is removed (cable still in place). Sometimes the PiJuice will go to a slow green blinking LED like I'd expect after sometime, other times it will stay solid green, but the red light on the Pi is still active regardless of whether there is external power applied or not. I incorrectly stated in my initial post that it automatically booted up in the morning after manual a halt - I have to manually press the switch for it to restart after power is restored (only when manual halting). To fully utilize the 'Software Halt Power Off' option, is it required to power the PiJuice directly with the microUSB or should this work with power being supplied by USB-C to the RPi? For reference, this is a Raspberry Pi 4 B with no pogo pin soldered in. I am running PiJuice firmware 1.5, software 1.7, on 32 bit

> Linux 5.10.17-v71+ #1414 SMP Fri Apr 30 13:20:47 BST 2021

I have tried this with halt, shutdown, poweroff, as well as trying a shutdown python script (made executable and owned by pi, etc) to mimic SYS_FUNC_HALT_POW_OFF, with little to no difference. To reiterate, this all works exactly as expected when it is a true power failure and the PiJuice fails over to batter power, where it safely shuts down and removes power, but manually halting does not. Are any other options needed or user type functions needed?

**Reply by mbstechnologiessales** [reply]
Found a reply you had on a previously closed Issue ticket from 2019 that stated: 
> The 'Software Halt Power off' setting takes care of item 4. When you shut down the Pi from the terminal or desktop the 5V power to the Pi will be removed after 20 sec.
> _**Also here cycling the external power will **not** boot the Pi.**_

So, assuming power cycling the external power will NOT boot the Pi after a 'Software Halt Power Off', even in the more recent firmware/software, then this would be an expected result in my setup. If this will not be possible to do after a manual halt, then I may just have to rethink manual shutdowns all together and come up with another plan. Manual halting will only be used on a few rare occasions, and the setup will work wonderfully for normal power failover and safe shutdowns.

**Reply by tvoverbeek** [reply]
Yes, you are right. In firmware 1.5 the power off delay gets cancelled when the external power is still present.

Here is a way your scenario could be implemented.
There is an external input IO2 on P3 which can be programmed as a digital input to wake up the Pi on a rising or falling edge.
If you connect your external 5V power input to the PiJuice micro-usb then you also have the external 5V available on the + pin of J4
since J4 is in parallel to the power connections in the micro-usb connector.
The IO2 ping is connected to an MCU GPIO which is 5V tolerant (MCU is running on 3.3V).
- Connect the + pin of J4 to the IO2 input.
- Program IO2 as digital input, pull-down, wakeup on rising edge.
- Do not enable wakeup-on-charge.

Now when using the 'normal' safe shutdown at night due to external power off the PiJuice powers the shiudown and after 60 seconds
the 5V to the Pi is removed (Pi red led goes off) and the PiJuice goes to low power mode, but he MCU remains running.
When external power appears we get a rising 5V edge on IO2 and the Pi starts.

When using a manual shutdown the 'Software Halt Power Off' option removes the 5V after the specified delay and then behaves
the same as above. When power is switched on in the morning the Pi will start again.

Tested it on my setup and for me it works.
See picture below:
![20210908_201532](https://user-images.githubusercontent.com/1197294/132563669-a416f216-6286-4b7b-8912-16b9cd4d95ef.jpg)
